guides/plugins/plugins/storefront/add-custom-page.md

```diff
@@ -2,19 +2,25 @@
 
 ## Overview
 
-In this guide you'll learn how to create a custom storefront page.
+In this guide you'll learn how to create custom page for your Storefront.
+A page in general consists of a controller, a page loader, a "page loaded" event and a page class, which is like a struct and contains
+most necessary data for the page.
 
 ## Prerequisites
 
-In order to add your own page service for your plugin, you first need a plugin as base. Therefore, you can refer to the [Plugin Base Guide](../plugin-base-guide.md).
+In order to add your own custom page for your plugin, you first need a plugin as base. Therefore, you can refer to the [Plugin Base Guide](../plugin-base-guide.md).
+Since you need to load your page with a controller, you might want to have a look at our guide about [creating a controller](./add-custom-controller.md) first.
+The controller created in the previously mentioned controller guide will also be used in this guide.
 
-Also having understood the concepts behind pages, pagelets and the pageloader [PLACEHOLDER-LINK: Page, pagelet, pageloader concept] will come in handy here.
+## Adding custom page
 
-## Adding custom storefront page
+In the following sections, we'll create each of the necessary classes one by one.
+The first one will be controller, whose creation is not going to explained here again. Have a look at the guide about [creating a controller](./add-custom-controller.md)
+to see why it works.
 
-### Storefront Controller Class example
+### Creating ExampleController
 
-First of all we have to create a new controller which extends from the `StorefrontController`. A controller is also just a service which can be registered via the DI-Container. Furthermore, we have to define our `RouteScope` via annotation, it is used to define which domain a route is part of and **needs to be set for every route**. In our case the scope is `storefront`. Now we can create an example function with a `Route` annotation which has to contain our route, in this case it's `/example`. Below you can find an example implementation where we render an `example.html.twig` template file with a template variable `example`.
+As already mentioned, the controller creation is not explained here, just an example controller.
```

Comment:
third time you mentioned that you are not going to explain it 😃

Reply:
And I'm not going to explain why I did that! :-P

Reply:
Fixed